addons/pos_details/static/src/js/reports.js

```diff
@@ -0,0 +1,63 @@
+odoo.define('pos_details.pos_details', function (require) {
+"use strict";
+
+var PosBaseWidget = require('point_of_sale.BaseWidget');
+var devices = require('point_of_sale.devices');
+var screens = require('point_of_sale.screens');
+var chrome = require('point_of_sale.chrome');
+var core = require('web.core');
+var Model = require('web.DataModel');
+
+var QWeb = core.qweb;
+var _t = core._t;
+
+
+var SaleDetailsButton = PosBaseWidget.extend({
+    template: 'SaleDetailsButton',
+    start: function(){
+        var self = this;
+        this.$el.click(function(){
+            self.pos.proxy.print_sale_details();
+        });
+    },
+});
+
+chrome.Chrome.include({
+    build_widgets: function () {
+
+        // add report widget to left of proxy widget
+        var proxy_status_index = _.findIndex(this.widgets, function (widget) {
+            return widget.name === "proxy_status";
+        });
+
+        this.widgets.splice(proxy_status_index, 0, {
+                'name': 'sale_details',
+                'widget': SaleDetailsButton,
+                'append': '.pos-rightheader',
+            });
+
+            this._super();
+        },
+});
+
+devices.ProxyDevice.include({
+    print_sale_details: function() { 
+        var self = this;
+        new Model('pos.order').call('get_sale_details').then(function(result){
+            var env = {
+                company: self.pos.company,
+                pos: self.pos,
+                products: result.products,
+                payments: result.payments,
+                taxes: result.taxes,
+                total_paid: result.total_paid,
+                date: (new Date()).toLocaleString(),
+            };
+            var report = QWeb.render('SaleDetailsReport', env);
+            self.pos.proxy.print_receipt(report);
+        })
+    },
+});
+
```

Comment:
return SaleDetailsButton at the end of the module, this will allow people to extend it.